### PR TITLE
Add Euler's Tour Code under Graph section

### PR DESCRIPTION
**Pull request by KritikaShrinivasan**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: graph/euler's_tour.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+2 3
+2 4
+
+output:
+1 2 3 2 4 2 1
+*/
+
+#include <iostream>
+#include <vector>
+using namespace std;
```
Comment (mishraabhinn) [inline graph/euler's_tour.cpp:21]: Please remove this.

File: graph/euler's_tour.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+input:
+4
+1 2
+2 3
+2 4
+
+output:
+1 2 3 2 4 2 1
+*/
+
```
Comment (mishraabhinn) [inline graph/euler's_tour.cpp:18]: Add namespaces

File: graph/euler's_tour.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    vis[u] = 1;
+    eulerTour[id++] = u;
+    for (auto it : adj[u]) {
+        if (!vis[it]) {
+            eulerTree(it, id);
+            eulerTour[id++] = u;
+        }
+    }
+}
+
```
Comment (mishraabhinn) [inline graph/euler's_tour.cpp:40]: Please add automated tests

File: graph/euler's_tour.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Test Case :
+input:
+4
+1 2
+2 3
+2 4
+
+output:
+1 2 3 2 4 2 1
+*/
```
Comment (mishraabhinn) [inline graph/euler's_tour.cpp:17]: **
 * @file
 * @brief Add one line description here. Should contain a Wikipedia
 * link or another source explaining the algorithm/implementation.
 * @details
 * This is a multi-line
 * description containing links, references,
 * math equations, etc.
 * @author [Name](https://github.com/handle)
 * @see related_file.cpp, another_file.cpp
 */

Comment (mishraabhinn) [reply graph/euler's_tour.cpp:17]: https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md
Please read this.

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

File: graph/euler's_tour.cpp (hunk 1 above)
Comment (KritikaShrinivasan) [reply graph/euler's_tour.cpp]: resolved. pls check.

File: graph/euler's_tour.cpp (hunk 2 above)
Comment (KritikaShrinivasan) [reply graph/euler's_tour.cpp]: resolved pls check.

File: graph/euler's_tour.cpp (hunk 3 above)
Comment (KritikaShrinivasan) [reply graph/euler's_tour.cpp]: resolved. pls check

File: graph/euler's_tour.cpp (hunk 4 above)
Comment (KritikaShrinivasan) [reply graph/euler's_tour.cpp]: resolved. pls check

File: graph/euler_tour.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+void solve(const std::vector<std::vector<size_t>> &adj, size_t root) {
+    size_t id = 0, n = adj.size();
+    std::vector<bool>vis(n, false);
+    eulerTree(adj, root, &vis, id);
+    for (int i = 0; i < (2 * n - 1); i++) 
+        std::cout << eulerTour[i] << " ";
+}
+} // namespace EulerTour
+
```
Comment (mishraabhinn) [inline graph/euler_tour.cpp:55]: Can you make an automated test for this? And verify the same using assert.

**Review by mishraabhinn — CHANGES_REQUESTED** [review]

File: graph/euler_tour.cpp (hunk 5 above)
Comment (KritikaShrinivasan) [reply graph/euler_tour.cpp:85]: Done. pls check

**Pull request closed without merging**